File: solved/18258/18258.cpp
#include <iostream>
#include <queue>
using namespace std;

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    int N;
    cin >> N;
    string s;
    queue<int> q;
    while (N--)
    {
        cin >> s;
        if (s == "push")
        {
            int n;
            cin >> n;
            q.push(n);
        }
        else if(s == "pop")
        {
            if (q.empty())
            {
                cout << -1 << '\n';
            }
            else
            {
                cout << q.front() << '\n';
                q.pop();
            }
        }
        else if(s == "size")
        {
            cout << q.size() << '\n';
        }
        else if(s == "empty")
        {
            cout << (q.empty() ? 1 : 0) << '\n';
        }
        else if(s == "front")
        {
            if (q.empty())
                cout << -1 << '\n';
            else
                cout << q.front() << '\n';
        }
        else if (s == "back")
        {
            if (q.empty())
                cout << -1 << '\n';
            else
                cout << q.back() << '\n';
        }
    }
    return 0;
}

File: solved/18258/18258.py
from collections import deque
import sys
input = sys.stdin.readline

N = int(input().strip())
q = deque()
for _ in range(N):
    s = list(input().strip().split())
    if s[0] == "push":
        q.append(int(s[1]))
    elif s[0] == "pop":
        if q:
            print(q.popleft())
        else:
            print(-1)
    elif s[0] == "size":
        print(len(q))
    elif s[0] == "empty":
        print(0 if q else 1)
    elif s[0] == "front":
        if q:
            print(q[0])
        else:
            print(-1)
    elif s[0] == "back":
        if q:
            print(q[-1])
        else:
            print(-1)